app/Model/ApiUser.php

```diff
@@ -153,16 +153,20 @@ public function beforeValidate($options = array())
     if(!empty($this->data['ApiUser'])) {
       // The username must begin with "co_<co_id>.".
       $prefix = "co_" . $this->data['ApiUser']['co_id'] . ".";
-      // Prepend the prefix to the username i got from post
-      $this->data['ApiUser']['username'] = $prefix . $this->data['ApiUser']['username'];
-
+      // Prepend the prefix to the username
+      if(strpos($prefix, $this->data['ApiUser']['username']) === false) {
```

Comment:
@boshrin good catch. Thank you. I leveraged the `strpos` function. We know that the prefix is missing if the strpos is false or the strpos is greater than 0, zero.

Reply:
> By skipping the check entirely I can remove the co_ prefix on an update and claim an existing API username since the uniqueness check is no longer fired.

Move the uniqueness check to a separate function and to the `$validate` array. It will always fire and will return a meaningful message to the user

>Instead, we should (1) always check that the API username is prefixed with co_#, and if it isn't insert the prefix (whether add or update) and (2) always check for uniqueness, but on update ignore any username returned by the find() if the record id matches $this->id.

Done. In the `beforeValidate` we construct the username and then return.